File: eCakeShop.Models/Korisnik.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCakeShop.Models
{
    public class Korisnik
    {
        public int KorisnikID { get; set; }
        public string Ime { get; set; }
        public string Prezime { get; set; }

        [DisplayName("Datum rodjenja")]
        public DateTime DatumRodjenja { get; set; }

        public string Email { get; set; }
        public string Telefon { get; set; }
        public string Lokacija => $"{Grad?.Naziv},{Drzava?.Naziv}";
        [Browsable(false)]
        public int GradID { get; set; }
        [Browsable(false)]
        public int DrzavaID { get; set; }
        [DisplayName("Korisnicko ime")]
        public string KorisnickoIme { get; set; }
        public string Uloge => string.Join(", ", KorisnikUlogas?.Select(x => x.Uloga?.Naziv)?.ToList());
        [Browsable(false)]
        public virtual Grad Grad { get; set; }
        [Browsable(false)]
        public virtual Drzava Drzava { get; set; }
        [Browsable(false)]
        public virtual ICollection<KorisnikUloga> KorisnikUlogas { get; set; }
    }
}

File: e_cakeshop/lib/models/korisnik.dart
import 'package:e_cakeshop/models/drzava.dart';
import 'package:e_cakeshop/models/grad.dart';
import 'package:json_annotation/json_annotation.dart';
part 'korisnik.g.dart';

@JsonSerializable()
class Korisnik {
  int? korisnikID;
  String? ime;
  String? prezime;
  String? korisnickoIme;
  DateTime? datumRodjenja;
  String? email;
  String? telefon;
  String? lokacija;
  int? gradID;
  int? drzavaID;
  Grad? grad;
  Drzava? drzava;

  Korisnik() {}

  factory Korisnik.fromJson(Map<String, dynamic> json) =>
      _$KorisnikFromJson(json);

  Map<String, dynamic> toJson() => _$KorisnikToJson(this);
}
